internal/utils/config.go

```diff
@@ -409,6 +409,12 @@ type (
 		DoubleConfirmChanges bool                     `toml:"double_confirm_changes"`
 		EnableConfirmations  bool                     `toml:"enable_confirmations"`
 		Template             map[string]emailTemplate `toml:"template"`
+		SmtpAdminEmail       string                   `toml:"smtp_admin_email"`
+		SmtpHost             string                   `toml:"smtp_host"`
+		SmtpPort             string                   `toml:"smtp_port"`
+		SmtpUser             string                   `toml:"smtp_user"`
+		SmtpPass             string                   `toml:"smtp_pass"`
+		SmtpSenderName       string                   `toml:"smtp_sender_name"`
```

Comment:
```suggestion
		Smtp                 smtp                     `toml:"smtp"`
```

I think this is worth refactoring into its own struct. For eg.

```golang
smtp struct {
		Host       string `toml:"host"`
		Port       string `toml:"port"`
		User       string `toml:"user"`
		Pass       string `toml:"pass"`
		AdminEmail string `toml:"admin_email"`
		SenderName string `toml:"sender_name"`
	}
```

Reply:
Done.

Here's a patch file if you want to use an int for the port:

```
From 046b8a70922a74cdb32e4ae109930fc9e9f99847 Mon Sep 17 00:00:00 2001
From: rhnaxifg4y <rhnaxifg4y@mouafik.fr>
Date: Tue, 18 Jun 2024 16:17:59 +0100
Subject: [PATCH 1/1] use int for smtp port

---
 internal/start/start.go                        | 4 ++--
 internal/utils/config.go                       | 2 +-
 internal/utils/templates/init_config.test.toml | 2 +-
 internal/utils/templates/init_config.toml      | 2 +-
 4 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/internal/start/start.go b/internal/start/start.go
index f56ec427..f5248995 100644
--- a/internal/start/start.go
+++ b/internal/start/start.go
@@ -429,8 +429,8 @@ EOF
 				return utils.InbucketId
 			}(),
 			"GOTRUE_SMTP_PORT=" + func() string {
-				if utils.Config.Auth.Email.Smtp.Port != "2500" {
-					return utils.Config.Auth.Email.Smtp.Port
+				if utils.Config.Auth.Email.Smtp.Port != 0 {
+					return strconv.Itoa(utils.Config.Auth.Email.Smtp.Port)
 				}
 				return strconv.Itoa(2500)
 			}(),
diff --git a/internal/utils/config.go b/internal/utils/config.go
index b155b881..54fe6bb8 100644
--- a/internal/utils/config.go
+++ b/internal/utils/config.go
@@ -417,7 +417,7 @@ type (
 
 	smtp struct {
 		Host       string `toml:"host"`
-		Port       string `toml:"port"`
+		Port       int    `toml:"port"`
 		User       string `toml:"user"`
 		Pass       string `toml:"pass"`
 		AdminEmail string `toml:"admin_email"`
diff --git a/internal/utils/templates/init_config.test.toml b/internal/utils/templates/init_config.test.toml
index f442df33..558e72bf 100644
--- a/internal/utils/templates/init_config.test.toml
+++ b/internal/utils/templates/init_config.test.toml
@@ -104,7 +104,7 @@ max_frequency = "1s"
 # [auth.email.smtp]
 # smtp_admin_email = "3ilbi6ee8c@gmail.com"
 # smtp_host = "smtp.sendgrid.net"
-# smtp_port = "587"
+# smtp_port = 587
 # smtp_user = "apikey"
 # smtp_pass = "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"
 # smtp_sender_name = "John Doe"
diff --git a/internal/utils/templates/init_config.toml b/internal/utils/templates/init_config.toml
index 7ea808e5..e94eb132 100644
--- a/internal/utils/templates/init_config.toml
+++ b/internal/utils/templates/init_config.toml
@@ -106,7 +106,7 @@ max_frequency = "1s"
 # [auth.email.smtp]
 # smtp_admin_email = "3ilbi6ee8c@gmail.com"
 # smtp_host = "smtp.sendgrid.net"
-# smtp_port = "587"
+# smtp_port = 587
 # smtp_user = "apikey"
 # smtp_pass = "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"
 # smtp_sender_name = "John Doe"
-- 
2.45.1.windows.1
```

That should fix the CI, but not sure why Inbucket doesn't work anymore.

> "error":"Error sending invite email: gomail: could not send email 1: gomail: invalid address \"\": mail: no address"